assets/js/modules/adsense/components/common/AdSenseConnectCTA.js

```diff
@@ -0,0 +1,241 @@
+/**
+ * AdSenseConnectCTA component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+import { ESCAPE } from '@wordpress/keycodes';
+import {
+	createInterpolateElement,
+	useEffect,
+	useCallback,
+	useState,
+} from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { MODULES_ADSENSE } from '../../datastore/constants';
+import { Grid, Row, Cell } from '../../../../material-components';
+import { ADSENSE_CTA_WIDGET_DISMISSED_ITEM_KEY } from '../../constants';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import { CORE_MODULES } from '../../../../googlesitekit/modules/datastore/constants';
+import { CORE_LOCATION } from '../../../../googlesitekit/datastore/location/constants';
+import Link from '../../../../components/Link';
+import Button from '../../../../components/Button';
+import Portal from '../../../../components/Portal';
+import Dialog from '../../../../components/Dialog';
+import AdSenseIcon from '../../../../../svg/adsense.svg';
+const { useSelect, useDispatch } = Data;
+
+export default function AdSenseConnectCTA() {
+	const [ inProgress, setInProgress ] = useState( false );
+	const [ dialogActive, setDialogActive ] = useState( false );
+
+	const { dismissItem } = useDispatch( CORE_USER );
+	const { navigateTo } = useDispatch( CORE_LOCATION );
+	const { activateModule } = useDispatch( CORE_MODULES );
+	const { setInternalServerError } = useDispatch( CORE_SITE );
+
+	const supportURL = useSelect( ( select ) =>
+		select( CORE_SITE ).getGoogleSupportURL( {
+			path: '/adsense/answer/9724',
+		} )
+	);
+	const adminReauthURL = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getAdminReauthURL()
+	);
+	const adSenseModuleActive = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleActive( 'adsense' )
+	);
+	const adSenseModuleConnected = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleConnected( 'adsense' )
+	);
+
+	const handleConnect = useCallback( async () => {
+		const { response, error } = await activateModule( 'adsense' );
```

Comment:
Per my reply in https://github.com/google/site-kit-wp/issues/4145#issuecomment-967287511, let's add a tracking event here.

Reply:
👍